src/v/cloud_storage/cache_service.cc

```diff
@@ -35,6 +35,7 @@
 #include <algorithm>
 #include <exception>
 #include <filesystem>
+#include <optional>
```

Comment:
nit: i think these commits should be merged in some way. The first commit isn't really do what's advertise and it will make it confusing upon browsing through git history

Reply:
>nit: i think these commits should be merged in some way. The first commit isn't really do what's advertise and it will make it confusing upon browsing through git history

agree. maybe they should be fully merged, but if they are to be kept separate, they need to have a cleaner separation with accurate commit messages.